tools/cjxl_ng_main.cc

```diff
@@ -235,58 +216,40 @@ DEFINE_int32(photon_noise, 0,
-    "Max. butteraugli distance, lower = higher quality. Range: 0 .. 25.\n"
+    distance, 1.0,
+    "Max. butteraugli distance, lower = higher quality.\n"
     "    0.0 = mathematically lossless. Default for already-lossy input "
     "(JPEG/GIF).\n"
     "    1.0 = visually lossless. Default for other input.\n"
-    "    Recommended range: 0.5 .. 3.0.");
-
-DEFINE_int64(target_size, 0,  // TODO(tfish): wire this up.
-             "Aim at file size of N bytes.\n"
-             "    Compresses to 1 % of the target size in ideal conditions.\n"
-             "    Runs the same algorithm as --target_bpp");
-
-DEFINE_double(target_bpp, 0.0,  // TODO(tfish): wire this up.
-              "Aim at file size that has N bits per pixel.\n"
-              "    Compresses to 1 % of the target BPP in ideal conditions.");
+    "    Recommended range: 0.5 .. 3.0. Mutually exclusive with --quality.");
 
 DEFINE_double(
-    quality, 100.0,  // TODO(tfish): wire this up.
+    quality, 100.0,
     "Quality setting (is remapped to --distance). Range: -inf .. 100.\n"
     "    100 = mathematically lossless. Default for already-lossy input "
-    "(JPEG/GIF).\n    Positive quality values roughly match libjpeg "
-    "quality.");
+    "(JPEG/GIF).\n"
+    "    Other input gets encoded as per --distance default.\n"
+    "    Positive quality values roughly match libjpeg quality.\n"
+    "    Mutually exclusive with --distance.");
 
-DEFINE_int64(
-    effort, 7,
-    // TODO(tfish): Clarify discrepancy with team:
-    // Documentation says default==squirrel(7) here:
-    // https://libjxl.readthedocs.io/en/latest/api_encoder.html#_CPPv424JxlEncoderFrameSettingId
-    // but enc_params.h has kFalcon=7.
-    "Encoder effort setting. Range: 1 .. 9.\n"
-    "    Default: 7. Higher number is more effort (slower).");
+DEFINE_int64(effort, 3,
+             "Encoder effort setting. Range: 1 .. 9.\n"
+             "     Higher number is more effort (slower).");
 
 DEFINE_int32(brotli_effort, 9,
              "Brotli effort setting. Range: 0 .. 11.\n"
              "    Default: 9. Higher number is more effort (slower).");
 
 DEFINE_string(frame_indexing, "",
+              // TODO(tfish): Add a more convenient vanilla alternative.
               "If non-empty, a string matching '^[01]*$'. If this string has a "
               "'1' in i-th position, then the i-th frame will be indexed in "
               "the frame index box.");
 
-DEFINE_string(
-    // TODO(tfish): Clarify with team whether changing from int-param to string
-    // is OK here.
-    colortransform, "",
-    "The color transform to use. Valid values are: '' (= \"use default\"), "
-    "'RGB', 'XYB', 'YCbCr'.");
-
-DEFINE_string(
-    mquality, "",  // TODO(tfish): Wire this up.
-    "[modular encoding] lossy 'quality', in the form luma_q[,chroma_q] "
-    "(100=lossless, lower is more lossy)");
+DEFINE_string(colortransform, "",
+              "The color transform used by the input. Valid values are: '' (= "
```

Comment:
It's a bit more complicated than that:
- `RGB` means: don't do any color transform (this is the default when doing lossless)
- `XYB` means: encoder will convert the RGB input to XYB (this is the default when doing lossy)
- `YCbCr` means: the input is already in YCbCr, the encoder will not do any color transform but it will signal in the frame header(s) that the data is YCbCr so it will get decoded correctly (this is the default when doing lossless JPEG recompression, and it would also be the thing to do when doing lossless compression from y4m input, if we want to support that).

To be honest, I don't think this parameter for cjxl is very useful, since doing anything different than the default will either lead to incorrect results, or to poor results. So I wouldn't object to just removing this parameter.

Reply:
Agree. Removed the colortransform flag related code.